SimExLite/PMICalculators/SimpleScatteringPMICalculator.py

```diff
@@ -0,0 +1,11147 @@
+""":module SimpleScatteringPMICalculator: Module that holds the SimpleScatteringPMICalculator class."""
+
+import h5py
+import numpy
+from pathlib import Path
+import shutil
+import sys
+import os
+import tempfile
+
+
+from libpyvinyl import BaseCalculator, CalculatorParameters
+from libpyvinyl.BaseData import DataCollection
+from SimExLite.SampleData import SampleData, ASEFormat
+from SimExLite.WavefrontData import WavefrontData, WPGFormat
+from SimExLite.PMIData import PMIData, XMDYNFormat
+
+
+class SimpleScatteringPMICalculator(BaseCalculator):
+    """:class SimpleScatteringPMICalculator: Class representing simple elastic scattering process."""
+
+    def __init__(
+        self,
+        name: str,
+        input: DataCollection,
+        output_keys: str = "PMI",
+        output_data_types=PMIData,
+        output_filenames: str = "PMI.h5",
+        instrument_base_dir="./",
+        calculator_base_dir="SimpleScatteringPMICalculator",
+        parameters=None,
+    ):
+        super().__init__(
+            name,
+            input,
+            output_keys,
+            output_data_types=output_data_types,
+            output_filenames=output_filenames,
+            instrument_base_dir=instrument_base_dir,
+            calculator_base_dir=calculator_base_dir,
+            parameters=parameters,
+        )
+
+    def init_parameters(self):
+        """Initialize with default parameters"""
+        parameters = CalculatorParameters()
+        num_steps = parameters.new_parameter(
+            "number_of_steps", comment="The number of time steps. Default = 1"
+        )
+        num_steps.value = 1
+
+        self.parameters = parameters
+
+    def __check_input_type(self):
+        # This check can be implemented in the BaseCalculator class
+        if not isinstance(self.input.to_list()[0], SampleData):
+            raise TypeError(
+                f"input[0] should be in SampleData type, instead of {type(self.input.to_list()[0])}"
+            )
+        if not isinstance(self.input.to_list()[1], WavefrontData):
+            raise TypeError(
+                f"input[1] should be in WavefrontData type, instead of {type(self.input.to_list[1])}"
+            )
+
+    def backengine(self):
+
+        # Prepare input files
+        self.__check_input_type()
+        Path(self.base_dir).mkdir(parents=True, exist_ok=True)
+        assert len(self.output_filenames) == 1
+        output_fn = str(Path(self.base_dir) / self.output_filenames[0])
+        sample_data = self.input.to_list()[0]
```

Comment:
Is self.input enforced (and need to be enforced) to be a collection of exactly 2 data? If yes, we could use unpacking here: 
```
sample_data, wavefront_data = self.input.to_list()
```
If no, should it be? For me it seems that a DataCollection with more than 2 items is erroneous.

Reply:
Yes, it will always require 2 datasets.
The suggestion is implemented.